main.tf

```diff
@@ -217,14 +217,46 @@ resource "aws_lb_listener" "this" {
     }
   }
 
+  routing_http_response_server_enabled                                = try(each.value.routing_http_response_server_enabled, false)
+  routing_http_response_strict_transport_security_header_value        = try(each.value.routing_http_response_strict_transport_security_header_value, null)
+  routing_http_response_access_control_allow_origin_header_value      = try(each.value.routing_http_response_access_control_allow_origin_header_value, null)
+  routing_http_response_access_control_allow_methods_header_value     = try(each.value.routing_http_response_access_control_allow_methods_header_value, null)
+  routing_http_response_access_control_allow_headers_header_value     = try(each.value.routing_http_response_access_control_allow_headers_header_value, null)
+  routing_http_response_access_control_allow_credentials_header_value = try(each.value.routing_http_response_access_control_allow_credentials_header_value, null)
+  routing_http_response_access_control_expose_headers_header_value    = try(each.value.routing_http_response_access_control_expose_headers_header_value, null)
+  routing_http_response_access_control_max_age_header_value           = try(each.value.routing_http_response_access_control_max_age_header_value, null)
+  routing_http_response_content_security_policy_header_value          = try(each.value.routing_http_response_content_security_policy_header_value, null)
+  routing_http_response_x_content_type_options_header_value           = try(each.value.routing_http_response_x_content_type_options_header_value, null)
+  routing_http_response_x_frame_options_header_value                  = try(each.value.routing_http_response_x_frame_options_header_value, null)
+
   load_balancer_arn        = aws_lb.this[0].arn
   port                     = try(each.value.port, var.default_port)
   protocol                 = try(each.value.protocol, var.default_protocol)
   ssl_policy               = contains(["HTTPS", "TLS"], try(each.value.protocol, var.default_protocol)) ? try(each.value.ssl_policy, "ELBSecurityPolicy-TLS13-1-2-Res-2021-06") : try(each.value.ssl_policy, null)
   tcp_idle_timeout_seconds = try(each.value.tcp_idle_timeout_seconds, null)
   tags                     = merge(local.tags, try(each.value.tags, {}))
+
+  lifecycle {
+    replace_triggered_by = [terraform_data.listeaner_replacement_trigger[each.key]]
+  }
 }
 
+resource "terraform_data" "listeaner_replacement_trigger" {
```

Comment:
why is this required?

Reply:
@bryantbiggs the provider doesnt update in place the values. It update only on create/recreate the listener.

Reply:
I'm not seeing that reflected in the implementation - it is checking for updates https://github.com/hashicorp/terraform-provider-aws/pull/40736

if what you are saying is true, the provider should specify `ForceNew: true` - we are not going to handle that here

Reply:
NP, i will wait for a fix https://github.com/hashicorp/terraform-provider-aws/issues/40986